### PR TITLE
Added HttpClientNoMatchException for HttpClientRouter

### LINKED ISSUE #44: Use HttpClientNotFoundException for HttpClientRouter
| Q | A |
| --- | --- |
| Bug? | yes (BC ?) |
| New Feature? | no |
#### Actual Behavior

Actually the HttpClientRouter use a RequestException which is not easily identifiable
#### Expected Behavior

Have an exception that pinpoint this specific use case (and we could profit from the exception added by ClientPool)

### PR DESCRIPTION
**Pull request by Nyholm**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | yes
| Deprecations?   | no
| Related tickets | fixes #44 
| Documentation   | 
| License         | MIT

### CODE REVIEW

File: src/Exception/HttpClientNoMatchException.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+use Http\Client\Exception\TransferException;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Thrown when a http client match in the HTTPClientRouter.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class HttpClientNoMatchException extends TransferException
```
Comment (dbu) [inline src/Exception/HttpClientNoMatchException.php]: should this be final?

**Comment by dbu** [comment]
there are tests asserting the RequestException that need to be adjusted.

**Comment by Nyholm** [comment]
Oh, thank you for fixing this PR for me.

**Comment by dbu** [comment]
felt it was easier than telling you what to do :-)

**Pull request merged**